### DISCUSSION TITLE
WebView client certificate authentication and configure SSL for HttpsClient

### DISCUSSION THREAD

**Discussion by prabhav-mehra** [dotnet/maui#10770] [Q&A]
Hi,

I was wondering if maui has access to one of the Android api KeyChain.choosePrivateKeyAlias() and how can we use it?

Context,
If a user goes to a website which requires PKI authentication using certificate stored in the keychain, the Android Keychain api shows a dialog box with all the certificate stored for user to select and authenticate. I was wondering if this is possible in .net maui app as well using Android api or any other api, I couldn’t find any.

Thanks.

**Comment by xtuzy** [comment]
dotnet-android have this api, https://learn.microsoft.com/en-us/dotnet/api/android.security.keychain.chooseprivatekeyalias?view=xamarin-android-sdk-12.
you need to google and know how to do it use java,  then convert java to c#.

**Comment by prabhav-mehra** [comment]
Thanks @xtuzy. I got that working. Pretty new to .net so I am confused if I can in any way configure the HttpsClient or the HttpsClientHandler with the certificate and privateKey instance I get from android keychain.

So for instance, I have been using the HttpsUrlConnection in java to make NetworkRequests. There you can configure SSL with the certificate and privateKey instance and make a post/get etc request. In .net/c# tho, there are different classes. I was wondering if we can configure SSL HttpsClient with the same certificate and privateKey we get from android to make NetworkRequests or can we map the HttpsUrlConnection to HttpsClient.

There is a workaround where I can use Custom implementation for different platform for making these Requests. But I would like to use only one( HttpsClient) to make requests while configuring it with platform specific certificates.
